src/main/java/bio/terra/cli/command/resources/Describe.java

```diff
@@ -2,35 +2,73 @@
 
 import bio.terra.cli.command.helperclasses.BaseCommand;
 import bio.terra.cli.command.helperclasses.FormatOption;
-import bio.terra.cli.context.CloudResource;
 import bio.terra.cli.service.WorkspaceManager;
+import bio.terra.workspace.model.ResourceDescription;
 import picocli.CommandLine;
-import picocli.CommandLine.Command;
 
 /** This class corresponds to the third-level "terra resources describe" command. */
-@Command(name = "describe", description = "Describe an existing controlled resource.")
+@CommandLine.Command(name = "describe", description = "Describe a resource.")
 public class Describe extends BaseCommand {
-
   @CommandLine.Option(
       names = "--name",
       required = true,
-      description = "The name of the resource, scoped to the workspace.")
+      description = "Name of the resource, scoped to the workspace.")
   private String name;
 
   @CommandLine.Mixin FormatOption formatOption;
 
-  /** Describe an existing controlled resource. */
+  /** Describe a resource. */
   @Override
   protected void execute() {
-    CloudResource resource =
-        new WorkspaceManager(globalContext, workspaceContext).getControlledResource(name);
+    ResourceDescription resource =
+        new WorkspaceManager(globalContext, workspaceContext).getResource(name);
     formatOption.printReturnValue(resource, Describe::printText);
   }
 
   /** Print this command's output in text format. */
-  private static void printText(CloudResource returnValue) {
-    OUT.println("Name: " + returnValue.name);
-    OUT.println("Type: " + returnValue.type);
-    OUT.println("Cloud Id: " + returnValue.cloudId);
+  public static void printText(ResourceDescription returnValue) {
+    OUT.println("Name:         " + returnValue.getMetadata().getName());
+    OUT.println("Description:  " + returnValue.getMetadata().getDescription());
+    OUT.println("Stewardship:  " + returnValue.getMetadata().getStewardshipType());
+
+    if (returnValue.getMetadata().getControlledResourceMetadata() != null) {
+      OUT.println(
+          "Access scope: "
+              + returnValue.getMetadata().getControlledResourceMetadata().getAccessScope());
+      OUT.println(
+          "Managed by:   "
+              + returnValue.getMetadata().getControlledResourceMetadata().getManagedBy());
+      if (returnValue
+              .getMetadata()
+              .getControlledResourceMetadata()
+              .getPrivateResourceUser()
+              .getUserName()
+          != null) {
+        OUT.println(
+            "Private user: "
+                + returnValue
+                    .getMetadata()
+                    .getControlledResourceMetadata()
+                    .getPrivateResourceUser()
+                    .getUserName());
+      }
+    }
+
```

Comment:
Seeing five similar `switch(ResourceType)` statements raised some concerns, but I thought about it a bit and can’t think of an alternative that saves much boilerplate.  Might be worth revisiting if/when the number of possible resource types grows above 3 or 4…

Reply:
Perhaps this is a second good reason to use CLI-defined objects? Interfaces or sub-classes seem like they'd get rid of most of these switch statements. (e.g. sub-classes override delete, checkAccess, printText methods).

Reply:
Noted this potential for code cleanup in [PF-729](https://broadworkbench.atlassian.net/browse/PF-729).